operator/pkg/manifest/installer.go

```diff
@@ -99,6 +99,18 @@ type deployment struct {
 
 var (
 	componentDependencies = componentNameToListMap{
+		name.PilotComponentName: {
+			name.PolicyComponentName,
+			name.TelemetryComponentName,
+			name.GalleyComponentName,
+			name.CitadelComponentName,
+			name.NodeAgentComponentName,
+			name.SidecarInjectorComponentName,
+			name.CNIComponentName,
+			name.IngressComponentName,
+			name.EgressComponentName,
+			name.AddonComponentName,
+		},
```

Comment:
why do we still need anything except pilot to directly depend on base in the list below?

Reply:
Good point, we can probably make it `base -> pilot -> everything else`